docs/content/en/docs/pipeline-stages/status-check.md

```diff
@@ -0,0 +1,108 @@
+{{% readfile file="samples/deployers/status-check.yaml" %}}
+
+With the `--status-check` flag, for each `Deployment` resource, `skaffold deploy` will wait for
+the time specified by [`progressDeadlineSeconds`](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#progress-deadline-seconds)
+from the deployment configuration.
+
+If the `Deployment.spec.progressDeadlineSeconds` is not set, Skaffold will either wait for
+the time specified in the `statusCheckDeadlineSeconds` field of the deployment config stanza in the `skaffold.yaml`, or
+default to 10 minutes if this is not specified.
+
+In the case that both `statusCheckDeadlineSeconds` and `Deployment.spec.progressDeadlineSeconds` are set, precedence
+is given to `Deployment.spec.progressDeadline` **only if it is less than** `statusCheckDeadlineSeconds`.
+
+For example, the `Deployment` below with `progressDeadlineSeconds` set to 5 minutes,
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: getting-started
+spec:
+  progressDeadlineSeconds: 300
+  template:
+    spec:
+      containers:
+      - name: cannot-run
+        image: gcr.io/k8s-skaffold/getting-started-foo
+```
+
+if the `skaffold.yaml` overrides the deadline to make sure deployment stabilizes in a 60 seconds,
+
+```yaml
+apiVersion: skaffold/v1
+deploy:
+  statusCheckDeadlineSeconds: 60
+  kubectl:
+    manifests:
+    - k8s-*
+```
+
+Running `skaffold deploy`
+
+```code
+skaffold deploy --status-check
+```
+will result in an error after waiting for 1 minute:
+
+```bash
+Tags used in deployment:
+Starting deploy...
+kubectl client version: 1.11+
+kubectl version 1.12.0 or greater is recommended for use with Skaffold
+ - deployment.apps/getting-started created
+Waiting for deployments to stabilize
+ - default:deployment/getting-started Waiting for rollout to finish: 0 of 1 updated replicas are available...
+ - default:deployment/getting-started failed. Error: received Ctrl-C or deployments could not stabilize within 1m: kubectl rollout status command interrupted.
+FATA[0006] 1/1 deployment(s) failed
+```
+
+### Configuring `status-check` for multiple deployers or multiple modules
+
+If you define multiple deployers, say `kubectl`, `helm`, and `kustomize`, all in the same skaffold config, or compose a multi-config project by importing other configs as dependencies, then the `status-check` can be run in one of two ways:
+- _Single status check after all deployers are run_. This is the default and it runs a single `status-check` at the end for resources deployed from all deployers across all skaffold configs.
+- _Per-deployer status check_. This can be enabled by using the `--iterative-status-check=true` flag. This will run a `status-check` iteratively after every individual deployer runs. This can be especially useful when there are startup dependencies between services, or you need to strictly enforce the time and order in which resources are deployed. 
```

Comment:
```suggestion
If you define multiple deployers, say `kubectl`, `helm`, and `kustomize`, all in the same skaffold config, or compose a multi-config project by importing other configs as dependencies, then the status check can be run in one of two ways:
- _Single status check after all deployers are run_. This is the default and it runs a single status check at the end for resources deployed from all deployers across all skaffold configs.
- _Per-deployer status check_. This can be enabled by using the `--iterative-status-check=true` flag. This will run a status check after every individual deployer runs. This can be especially useful when there are startup dependencies between services, or you need to strictly enforce the time and order in which resources are deployed. 
```